src/shared/install.c

```diff
@@ -1757,31 +1787,17 @@ static int install_info_symlink_alias(
         assert(config_path);
 
         STRV_FOREACH(s, i->aliases) {
-                _cleanup_free_ char *alias_path = NULL, *dst = NULL;
+                _cleanup_free_ char *alias_path = NULL, *dst = NULL, *dst2 = NULL;
 
                 q = install_full_printf(i, *s, &dst);
                 if (q < 0)
                         return q;
 
-                q = unit_file_verify_alias(i, dst);
+                q = unit_file_verify_alias(i, dst, &dst2);
```

Comment:
Maybe naming `dst2` something like `dst_normalized` would make it easier to understand what is going on here. Also, maybe `unit_file_verify_alias()` could always return the destination unit name, not only when it requires tweaking.

Reply:
I did it this way to avoid an allocation. It also gives the caller clear information if the target name changed. There is most likely going to be just a handful of callers of this, so I think it's OK to leave the interface a bit more complex like this.

I'll change the name.